tools/source_updater/updater/source.py

```diff
@@ -0,0 +1,164 @@
+import json
+import os
+import yaml
+from pydantic.error_wrappers import ValidationError
+from typing import Set
+
+from airbyte_cdk.connector import BaseConnector
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from updater.config import Config
+
+# This is needed as when interpreting SecretsManager, the path used is determined by the VERSION environment variable
+os.environ["VERSION"] = "dev"
+from ci_credentials import SecretsManager  # noqa: E402
+
+
+class SourceRepository:
+    def exists(self, source_name: str) -> bool:
+        return os.path.exists(self._path_from_name(source_name))
+
+    def fetch_catalog(self, source_name: str) -> ConfiguredAirbyteCatalog:
+        try:
+            return ConfiguredAirbyteCatalog.parse_obj(BaseConnector.read_config(self._catalog_path(source_name)))
+        except ValidationError as error:
+            raise ValueError(error)
+
+    def update_catalog(self, source_name: str, catalog: ConfiguredAirbyteCatalog) -> None:
+        with open(self._catalog_path(source_name), "w") as catalog_file:
+            json.dump(catalog.dict(exclude_none=True), catalog_file, indent=2, default=lambda x: x.value)
```

Comment:
This is still causing some diffs as our lists are not always set up the same way. For example:
* [Inline](https://github.com/airbytehq/airbyte/blob/67a01d6be2f5064456c6c53b87301dafcd465153/airbyte-integrations/connectors/source-prestashop/integration_tests/configured_catalog.json#L7)
* [Not inline](https://github.com/airbytehq/airbyte/blob/67a01d6be2f5064456c6c53b87301dafcd465153/airbyte-integrations/connectors/source-prestashop/integration_tests/configured_catalog.json#L2)

However, the amount of diff is very minimal

Reply:
are the links pointing to the right line? I don't understand what diff you're referring to

Reply:
The concern is to have as few diffs as possible when updating the catalog. Using `json.dump(catalog.dict(exclude_none=True), catalog_file, indent=2, default=lambda x: x.value)` gives somewhat good results but it is not perfect because some lists are inline and others are not (see the link in the initial comment)